### PR TITLE
GameActionUtil: store OptionalKeywordAmount in CastSA

### PR DESCRIPTION
**Pull request by Hanmac**
@tool4ever i moved the amount from the trigger object, into the CastSA info

might need a better name for "CastSAOptKeyword"

### CODE REVIEW

**Comment by Hanmac** [comment]
@tehdiplomat your opinion on the Name of the Functions?

File: forge-gui/res/cardsfolder/o/ob_nixilis_the_adversary.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@ Name:Ob Nixilis, the Adversary
 ManaCost:1 B R
 Types:Legendary Planeswalker Nixilis
 Loyalty:3
-K:Casualty:X:NonLegendary$ True | SetLoyalty$ Casualty:The copy isn't legendary and has starting loyalty X.
+K:Casualty:X:NonLegendary$ True | SetLoyalty$ CasualtyPaid:The copy isn't legendary and has starting loyalty X.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/o/ob_nixilis_the_adversary.txt:5]: broken now for X=0

Comment (Hanmac) [reply forge-gui/res/cardsfolder/o/ob_nixilis_the_adversary.txt]: fixed

**Comment by Hanmac** [comment]
@tool4ever can you help me find a better name for the attribute and functions?

**Comment by tool4ever** [comment]
well `OptionalGenericCostPaid` already exists
so maybe `OptionalKeywordCostPaid[Times]`?

**Comment by Hanmac** [comment]
@tool4ever I got some better idea for this MR

if I extend the Map into a Table using Keyword as additional Key like this:
`Map<Pair<Long, Long>, Integer>` → `Table<Keyword, Pair<Long, Long>, Integer>`

then I can use this for other cases like Multikicker too

**Comment by Hanmac** [comment]
@tool4ever I renamed my stuff and used a Table for future uses (like Multikicker)

File: forge-game/src/main/java/forge/game/card/CardFactoryUtil.java [hunk 2]
```diff
@@ -907,13 +907,16 @@ public static void addTriggerAbility(final KeywordInterface inst, final Card car
...
+                abString += " | " + k[2];
             }
 
             final Trigger casualtyTrigger = TriggerHandler.parseTrigger(trigScript, card, intrinsic);
-            casualtyTrigger.setOverridingAbility(AbilityFactory.getAbility(abString, card));
-            casualtyTrigger.setSVar("Casualty", "0");
-            casualtyTrigger.setSVar("CasualtyPaid", "0");
+            SpellAbility sa = AbilityFactory.getAbility(abString, card);
+            sa.setSVar("CasualtyPaid", "Count$hasOptionalKeywordAmount");
+            sa.setSVar("Casualty", "Count$OptionalKeywordAmount");
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/card/CardFactoryUtil.java:916]: the trigger should only need the first SVar and the sa only the second when inside that if part?

Comment (Hanmac) [reply forge-game/src/main/java/forge/game/card/CardFactoryUtil.java:916]: probably, should i change that, or should i keep it? i mean the extra doesn't hurt?

**Comment by Hanmac** [comment]
also on my local branch, i testing out Multikicker for Human, it works

AI might need more love

**Review by tool4ever — APPROVED** [review]

**Pull request merged**